src/m365/spo/commands/tenant/tenant-applicationcustomizer-remove.ts

```diff
@@ -0,0 +1,187 @@
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id) {
+          const id: number = parseInt(args.options.id);
+          if (isNaN(id)) {
+            return `${args.options.id} is not a valid list item ID`;
+          }
+        }
+
+        if (args.options.clientSideComponentId &&
+          !validation.isValidGuid(args.options.clientSideComponentId as string)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (args.options.confirm) {
+      await this.removeTenantApplicationCustomizer(logger, args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the tenant applicationcustomizer ${args.options.id || args.options.title || args.options.clientSideComponentId}?`
+      });
+
+      if (result.continue) {
+        await this.removeTenantApplicationCustomizer(logger, args);
+      }
+    }
+  }
+
+  public async getTenantApplicationCustomizer(logger: Logger, args: CommandArgs, requestUrl: string): Promise<number> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the tenant application customizer ${args.options.id || args.options.title || args.options.clientSideComponentId}`);
+    }
+
+    const filter: string[] = [`TenantWideExtensionLocation eq 'ClientSideExtension.ApplicationCustomizer'`];
+    if (args.options.title) {
+      filter.push(`Title eq '${args.options.title}'`);
+    }
+    else if (args.options.id) {
+      filter.push(`Id eq '${args.options.id}'`);
+    }
+    else if (args.options.clientSideComponentId) {
+      filter.push(`TenantWideExtensionComponentId eq '${args.options.clientSideComponentId}'`);
+    }
+
+    const reqOptions: CliRequestOptions = {
+      url: `${requestUrl}/items?$filter=${filter.join(' and ')}`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const listItemInstances: ListItemInstanceCollection = await request.get<ListItemInstanceCollection>(reqOptions);
+
+    if (listItemInstances.value.length === 0) {
+      throw 'The specified application customizer was not found';
+    }
+
+    if (listItemInstances.value.length > 1) {
+      throw `Multiple application customizers with ${args.options.title || args.options.clientSideComponentId} were found. Please disambiguate (IDs): ${listItemInstances.value.map(item => item.Id).join(', ')}`;
+    }
+
+    return listItemInstances.value[0].Id;
+
+  }
+
+  private async removeTenantApplicationCustomizer(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
```

Comment:
Let's move the try catch statement to the commandAction method, where it usually is.